snapshot-labs/delegate-registry-api
# feat: compute registry data

Closes: https://github.com/snapshot-labs/sx-monorepo/issues/485
Depends on: https://github.com/snapshot-labs/snapshot.js/pull/1050

This PR adds compute function that creates delegation data for delegateRegistryV1.

### Test plan

1. Run this project
2. Run [this query](http://localhost:3000/?variables=%7B%0A%20%20%22id%22%3A%20%22apecoin.eth%22%0A%7D&query=query%20Test(%24id%3A%20String!)%20%7B%0A%20%20governance(id%3A%20%24id)%20%7B%0A%20%20%20%20id%0A%20%20%20%20currentDelegates%0A%20%20%20%20totalDelegates%0A%20%20%20%20delegatedVotes%0A%20%20%20%20delegatedVotesRaw%0A%20%20%7D%0A%20%20delegates%20(where%3A%20%7B%20governance%3A%20%24id%7D)%20%7B%0A%20%20%20%20id%0A%20%20%20%20delegatedVotes%0A%20%20%20%20tokenHoldersRepresentedAmount%0A%20%20%7D%0A%7D%0A).
3. It will return empty response (because we don't wait for data to be computed, only return stored value).
4. See in logs that computation was started, wait for it to finish.
5. Repeat the request.
6. You see values and they are correct (compare with https://snapshot.org/#/delegate/apecoin.eth)

Query:
```gql
query Test($id: String!) {
  governance(id: $id) {
    id
    currentDelegates
    totalDelegates
    delegatedVotes
    delegatedVotesRaw
  }
  delegates (where: { governance: $id}) {
    id
    delegatedVotes
    tokenHoldersRepresentedAmount
  }
}

```
